core/src/main/java/org/infinispan/cache/impl/InvocationHelper.java

```diff
@@ -238,7 +248,7 @@ private Transaction getOngoingTransaction() {
             throw CompletableFutures.asCompletionException(e);
          }
          return result;
-      });
+      }, blockingExecutor), nonBlockingExecutor, ctx).toCompletableFuture();
```

Comment:
Not all implementations of `InvocationContext` have a proper `toString()`, maybe `ctx.getLockOwner()` instead?